### PR TITLE
[CELEBORN-1547] Worker#listTopDiskUseApps should return celeborn.metrics.app.topDiskUsage.count applications

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

`Worker#listTopDiskUseApps` should return `celeborn.metrics.app.topDiskUsage.count` applications.

### Why are the changes needed?

`Worker#listTopDiskUseApps` returns 2x `celeborn.metrics.app.topDiskUsage.count` application, which is not same as `celeborn.metrics.app.topDiskUsage.count` configuration at present. Meanwhile, `Worker#listTopDiskUseApps` reuses `StorageManager#topAppDiskUsage` method to get the top application list used for estimated application disk usage. Therefore, `Worker#listTopDiskUseApps` should return celeborn.metrics.app.topDiskUsage.count applications.

```
$ curl http://celeborn-worker:9096/listTopDiskUsedApps|grep used|wc -l
100
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

```
$ curl http://celeborn-worker:9096/listTopDiskUsedApps|grep used|wc -l
50
```

### CODE REVIEW

**Comment by SteNicholas** [comment]
Ping @FMX, @AngersZhuuuu.

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala [hunk 1]
```diff
@@ -525,7 +527,8 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
     hashSet
   }
 
-  def topAppDiskUsage: util.Map[String, Long] = {
+  def topAppDiskUsage(estimated: Boolean = false): util.Map[String, Long] = {
```
Comment (FMX) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala:530]: IMO, reportToMaster will be easier to understand.

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**